src/Compiler.ts

```diff
@@ -113,32 +132,41 @@ function removeInvalidOptionsFromArgv(commandLine: ReadonlyArray<string>, invali
 // Polyfill for report diagnostics
 function logError(commandLine: ts.ParsedCommandLine) {
     if (commandLine.errors.length !== 0) {
+        let invalidErrorCount = 0;
         commandLine.errors.forEach((err) => {
-            console.log(err.messageText);
+            // Allow all compiler options
+            if (err.code == 5023) {
+                invalidErrorCount += 1;
+            }
+            else {
+                console.log(err.messageText);
+            }
         });
-        process.exit(1);
+        if (invalidErrorCount != commandLine.errors.length) {
+            process.exit(1);
+        }
     }
 }
 
 function executeCommandLine(args: ReadonlyArray<string>) {
     // Right now luaTarget, version and help are the only cli options, if more are added we should
     // add a more advanced custom parser
-    var argv = minimist(args, {
+    const argv = minimist(args, {
         string: ['l'],
         alias: { h: 'help', v: 'version', l: 'luaTarget' },
```

Comment:
Add CLI options for addHeader and lualiboutdir. Don't forget to update help text and readme.